test/gw-ci/CMakeLists.txt

```diff
@@ -18,78 +48,151 @@ function(add_cycling_tests pslot YAML_PATH HOMEgfs RUNTESTS PROJECT_SOURCE_DIR T
-  message(STATUS "Tasks ${TASK_LIST}")
-
-  foreach(task ${TASK_LIST})
-    message(STATUS "preparing the full cycle ${task} for ${pslot} ctest")
-    add_test(NAME ${test_name}_${task}_${FULL_CYCLE}
-             COMMAND /bin/bash -c "${PROJECT_SOURCE_DIR}/test/gw-ci/run_exp.sh ${pslot} ${task} ${FULL_CYCLE}"
-             WORKING_DIRECTORY ${RUNTESTS})
-    set_tests_properties(${test_name}_${task}_${FULL_CYCLE} PROPERTIES LABELS "manual")
+  message(STATUS "Full-cycle tasks ${FULL_CYCLE_TASKS}")
+  foreach(task_name ${FULL_CYCLE_TASKS})
+    add_task(${task_name} ${test_prefix} ${FULL_CYCLE} ${pslot} ${RUNTESTS} ${PROJECT_SOURCE_DIR})
   endforeach()
 endfunction()
 
 if (WORKFLOW_TESTS)
   # Setup the environement
   set(HOMEgfs ${CMAKE_SOURCE_DIR}/../../..)
   set(RUNTESTS ${CMAKE_CURRENT_BINARY_DIR}/../../test/gw-ci)
+  
+  # GSI Atm DA C96/C48
+  # ------------------
+  set(pslot "C96C48_hybatmDA")
+  set(YAML_PATH ${HOMEgfs}/ci/cases/pr/${pslot}.yaml)
+  set(HALF_CYCLE_TASKS
+    "gdas_stage_ic"
+    "gdas_fcst"
+    "gdas_atmos_prod"
+    "enkfgdas_stage_ic"
+    "enkfgdas_fcst"
+    "enkfgdas_echgres"
+    "enkfgdas_epmn")
+  set(FULL_CYCLE_TASKS
+    "gdas_prep"
+    "gdas_anal"
+    "gdas_sfcanl"
+    "gdas_analcalc"
+    "gdas_fcst"
+    "enkfgdas_eobs"
+    "enkfgdas_ediag"
+    "enkfgdas_eupd"
+    "enkfgdas_ecmn"
+    "enkfgdas_esfc"
+    "enkfgdas_fcst"
+  )
 
-  # 3DVAR WCDA, low-res
+  add_cycling_tests(${pslot} ${YAML_PATH} ${HOMEgfs} ${RUNTESTS} ${PROJECT_SOURCE_DIR} "${HALF_CYCLE_TASKS}" "${FULL_CYCLE_TASKS}")
+  # JEDI Atm DA C96/C48
   # -------------------
-  set(pslot "WCDA-3DVAR-C48mx500")
-  set(YAML_PATH ${HOMEgfs}/ci/cases/pr/C48mx500_3DVarAOWCDA.yaml)
-  set(TASK_LIST
+  set(pslot "C96C48_ufs_hybatmDA")
+  set(YAML_PATH ${HOMEgfs}/ci/cases/pr/${pslot}.yaml)
+  set(HALF_CYCLE_TASKS
+    "gdas_stage_ic"
+    "gdas_fcst"
+    "gdas_atmos_prod"
+    "enkfgdas_stage_ic"
+    "enkfgdas_fcst"
+    "enkfgdas_echgres"
+    "enkfgdas_epmn")  
+  set(FULL_CYCLE_TASKS
+    "gdas_prep"
+    "gdas_prepatmiodaobs"
+    "gdas_atmanlinit"
+    "gdas_atmanlvar"
+    "gdas_atmanlfv3inc"
+    "gdas_atmanlfinal"
+    "gdas_sfcanl"
+    "gdas_analcalc"
+    "gdas_fcst"
+    "enkfgdas_atmensanlinit"
+    "enkfgdas_atmensanlobs"
+    "enkfgdas_atmensanlsol"
+    "enkfgdas_atmensanlfv3inc"
+    "enkfgdas_atmensanlfinal"
+    "enkfgdas_ecen000" # No IAU yet, so only one task
+    "enkfgdas_esfc"
+    "enkfgdas_fcst"
+  )
+  add_cycling_tests(${pslot} ${YAML_PATH} ${HOMEgfs} ${RUNTESTS} ${PROJECT_SOURCE_DIR} "${HALF_CYCLE_TASKS}" "${FULL_CYCLE_TASKS}")  
+
+  # Aero-Land DA C96
+  # ----------------
+  set(pslot "C96C48_hybatmaerosnowDA")
+  set(YAML_PATH ${HOMEgfs}/ci/cases/pr/${pslot}.yaml)
+  set(HALF_CYCLE_TASKS
+    "gdas_stage_ic"
+    "gdas_fcst"
+    "gdas_atmos_prod"
+    "gdas_aeroanlgenb"
+    "enkfgdas_stage_ic"
+    "enkfgdas_fcst"
+    "enkfgdas_echgres"
+    "enkfgdas_epmn")  
+  set(FULL_CYCLE_TASKS
```

Comment:
@CoryMartin-NOAA Yeah, I thought about that. Could probably do the same thing ```setup_xml.py``` does to check dependencies. The problem is that I want to skip certain jobs like archiving.

Reply:
a list of jobs to exclude, but otherwise it checks dependencies and includes all other jobs? or is that pain not worth it? I just fear that we will be doing a lot of maintenance on this file in the future to keep it up to date

Reply:
I think maybe this can be an iterative process. I have another PR planned to test the memory and time usage, so maybe I can add it there. It will be a bit of a pain because I'll have to interface the CMake and Python

Reply:
Keep it simple and definitely don't add unnecessary obfuscation to the code! We've been using this for a while and the maintenance required was trivial.

Reply:
Yeah, I mean we'll know if something breaks when the tests stop passing